scalafmt-cli/src/main/scala/org/scalafmt/cli/Cli.scala

```diff
@@ -69,20 +69,20 @@ object Cli {
       .map(CliOptions.auto)
   }
 
-  private def expandArguments(args: Array[String]): Array[String] = {
-    args.flatMap {
-      case FileArgument(xs) => xs
-      case x => List(x)
-    }
-  }
-  private object FileArgument {
-    def unapply(arg: String): Option[Iterator[String]] = {
+  private def expandArguments(args: Seq[String]): Seq[String] = {
+    val builder = Seq.newBuilder[String]
+    args.foreach { arg =>
       val atFile = arg.stripPrefix("@")
-      if (atFile == arg) None // doesn't start with @
-      else if (atFile == "-") Some(Source.stdin.getLines())
-      else if (!Files.isRegularFile(Paths.get(atFile))) None
-      else Some(Source.fromFile(atFile).getLines())
+      if (atFile == arg) builder += arg // doesn't start with @
+      else if (atFile == "-") builder ++= Source.stdin.getLines()
+      else if (!Files.isRegularFile(Paths.get(atFile))) builder += arg
+      else {
+        val src = Source.fromFile(atFile)
+        try builder ++= src.getLines()
+        finally src.close()
```

Comment:
You could also use `scala.util.Using` 
```suggestion
        builder ++= Using.resource(Source.fromFile(atFile))(_.getLines())
```

Reply:
done, thank you, didn't know about this

Reply:
it's only available for scala 2.13. added this package for 2.12:

https://index.scala-lang.org/bigwheel/util-backports/util-backports/2.1